advanced_alchemy/mixins/unique.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from sqlalchemy import ColumnElement, select
+
+if TYPE_CHECKING:
+    from collections.abc import Hashable
+
+    from sqlalchemy.ext.asyncio import AsyncSession
+    from sqlalchemy.ext.asyncio.scoping import async_scoped_session
+    from sqlalchemy.orm import Session
+    from sqlalchemy.orm.scoping import scoped_session
+    from typing_extensions import Self
+
+
+class UniqueMixin:
```

Comment:
I ended up not writing two different mixins (sync and async) as I felt doing so would make the user "lock in" to "sync only" or "async only" flows. If one wants to support both(sync and async), one would intuitively try to inherit from both the mixins and I was not sure if that would play well with the "inheritance chain".

So I added two `as_unique` methods that can coexist together.

Reply:
Yeh, I agree with this. ORM types are otherwise totally interchangeable between sync/async paradigms so it would be a shame to introduce some division down those lines here.